internal/controllers/nmc_reconciler.go

```diff
@@ -429,33 +484,15 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 		logger.Info("Processing worker Pod")
 
 		status := nmc.FindModuleStatus(nmcObj.Status.Modules, modNamespace, modName)
-		if status == nil {
-			status = &kmmv1beta1.NodeModuleStatus{
-				ModuleItem: kmmv1beta1.ModuleItem{
-					Name:      modName,
-					Namespace: modNamespace,
-				},
-			}
-		}
-
-		if irsName, err := getImageRepoSecretName(&p); err != nil {
-			logger.Info(
-				utils.WarnString("Error while looking for the imageRepoSecret volume"),
-				"error",
-				err,
-			)
-		} else if irsName != "" {
-			status.ImageRepoSecret = &v1.LocalObjectReference{Name: irsName}
-		}
-
-		status.ServiceAccountName = p.Spec.ServiceAccountName
-
-		updateModuleStatus := false
 
 		switch phase {
+		case v1.PodRunning:
+			// Delete Pod if orphan
+			if !specEntries.Has(types.NamespacedName{Namespace: modNamespace, Name: modName}) && status == nil {
```

Comment:
not sure i understand the handling of Running Pod: in case spec is missing we delete the running pod? What if it the Unload pod? Why do we want to delete it?

Reply:
It cannot be an unloader Pod because we check `status == nil`. This is to handle cases where we have a restarting loading Pod but the Module was deleted - and so was the spec entry.